packages/block-editor/src/hooks/pattern.js

```diff
@@ -172,37 +64,11 @@ function extendUsesContext( settings ) {
 const createEditFunctionWithPatternSource = () =>
 	createHigherOrderComponent(
 		( BlockEdit ) =>
-			( { name, attributes, setAttributes, context, ...props } ) => {
-				if ( ! context.patternId ) {
-					return (
-						<BlockEdit
-							name={ name }
-							context={ context }
-							attributes={ attributes }
-							setAttributes={ setAttributes }
-							{ ...props }
-						/>
-					);
-				}
-
-				const registry = useRegistry();
-				const {
-					attributes: updatedAttributes,
-					setAttributes: updatedSetAttributes,
-				} = useSource( { name, attributes, setAttributes, context } );
+			( { attributes, ...props } ) => {
+				const sourceAttributes = useSourceAttributes( attributes );
 
 				return (
-					<BlockEdit
-						name={ name }
-						context={ context }
-						attributes={ updatedAttributes }
-						setAttributes={ ( newAttributes ) =>
-							registry.batch( () =>
-								updatedSetAttributes( newAttributes )
-							)
-						}
-						{ ...props }
-					/>
+					<BlockEdit { ...props } attributes={ sourceAttributes } />
```

Comment:
This is interesting. I wonder if the override for attributes should also happen on the store level, similar to how modifications get applied to `updateBlockAttributes`. Otherwise, we risk that folks could omit using `attributes` passed to `Edit` and get local values for dynamic content when accessing the store directly similar to what would happen when they use `updateBlockAttributes`.